### ISSUE TITLE
no clean optimized and fast way to sort StringArray

### ISSUE THREAD

**Issue by SuperUserNameMan** [godotengine/godot#5202]
**Operating system or device - Godot version:**
official 2.0.3stable win64

**Issue description** (what happened, and what was expected):

Currently, the only way to sort a `StringArray` is to convert it back into an `Array` to `.sort()` it, then convert it back to a new `StringArray`.

Maybe it would be more optimized to have `StringArray.sort()` and `StringArray.sort_custom()` methods ?

**Steps to reproduce:**
**Link to minimal example project** (optional but very welcome):

```
var very_long_list = StringArray();
var temp = Array( very_long_list );
temp.sort();
very_long_list = StringArray( temp );
```

**Comment by Calinou** [comment]
This issue still seems to be relevant as of https://github.com/godotengine/godot/commit/b030834b6384d5f57e5fc63db21b643adc86b3da. Only Array has `sort()` and `sort_custom()` methods, not pooled arrays. I'm not sure if adding sorting methods to pooled arrays is desirable though.

**Comment by akien-mga** [comment]
Fixed by #55650.
